zerver/middleware.py

```diff
@@ -655,3 +663,36 @@ def should_redirect_with_slash(self, request: HttpRequest) -> bool:
         if settings.RUNNING_INSIDE_TORNADO:
             return False
         return super().should_redirect_with_slash(request)
+
+
+def validate_scim_bearer_token(request: HttpRequest) -> Optional[UserProfile]:
+    subdomain = get_subdomain(request)
+    valid_bearer_token, scim_bot_email = settings.SCIM_BEARER_TOKENS[subdomain]
+    # We really don't want a misconfiguration where this is unset,
+    # allowing free access to the SCIM API:
+    assert valid_bearer_token
+
+    if request.headers.get("Authorization") != f"Bearer {valid_bearer_token}":
+        return None
+
+    request_notes = RequestNotes.get_notes(request)
+    assert request_notes.realm
+    return get_user_by_delivery_email(scim_bot_email, request_notes.realm)
+
+
+class ZulipSCIMAuthCheckMiddleware(SCIMAuthCheckMiddleware):
+    def process_request(self, request: HttpRequest) -> Optional[HttpResponse]:
+        if self.should_log_request(request):
+            self.log_request(request)
```

Comment:
This, in the superclass, accesses `request.body` for logging purposes which I believe can be a problem with `POST` requests in Django. So this is a `TODO` to determine whether this is okay or needs to be overriden.